.github/workflows/main.yml

```diff
@@ -1,119 +1,127 @@
+        uses: biomejs/setup-biome@v2
+        with:
+          version: latest
+
+      - name: "unit(test): perform lints and formatting checks"
+        run: biome ci .
+
+      - name: "unit(test): perform check of typings"
+        run: npm run check
+
+      - name: "unit(test): perform unit test checks"
+        run: npm test
+
+      - name: "unit(test): upload coverage to CodeCov"
+        uses: codecov/codecov-action@v4.5.0
+        with:
+          directory: ./coverage
+          token: ${{ secrets.CODECOV_TOKEN }}
+
+      - name: "integration(botToken): post a message to channel"
+        id: slackToken
+        uses: ./
+        with:
+          method: chat.postMessage
+          token: ${{ secrets.SLACK_BOT_TOKEN }}
+          payload: |
+            channel: ${{ secrets.SLACK_CHANNEL_ID }}
+            text: "CI Post from slack-send GitHub Action! Succeeded!!"
+
+      - name: "integration(botToken): confirm a message was posted"
+        run: test -n "${{ steps.slackToken.outputs.ts }}"
+
+      - name: "integration(botToken): post a threaded response"
+        id: slackThreadResponse
+        uses: ./
+        with:
+          method: chat.postMessage
+          token: ${{ secrets.SLACK_BOT_TOKEN }}
+          payload: |
+            channel: ${{ secrets.SLACK_CHANNEL_ID }}
+            text: "This message should be posted as a response in thread"
+            thread_ts: ${{ steps.slackToken.outputs.thread_ts }}
+
+      - name: "integration(botToken): confirm a response was posted"
+        run: test -n "${{ steps.slackThreadResponse.outputs.ts }}"
+
+      - name: "integration(wfb): save the push event trigger commit URL"
+        if: "contains(github.event_name, 'push')"
+        run: |
+          url=${{ github.event.head_commit.url }}
+          echo "URL=$url" >> "$GITHUB_ENV"
+
+      - name: "integration(wfb): save the pull request event trigger commit URL"
+        if: "contains(github.event_name, 'pull_request')"
+        run: |
+          url=${{ github.event.pull_request.url }}
+          echo "URL=$url" >> "$GITHUB_ENV"
+
+      - name: "integration(wfb): send a payload via workflow builder webhook"
+        id: slackWorkflow
+        uses: ./
+        with:
+          # Workflow builder webhooks need to know the name of the keys in the payload in advance. Without normalizing, the github context payload keys can differ based on the GitHub trigger event type
+          # Normalized payload with info pulled out from GitHub trigger event
+          payload: '{"author":"${{ github.event.sender.login }}","url":"${{ env.URL}}", "repoName":"${{ github.event.repository.full_name }}", "status":"${{ job.status }}"}'
+          webhook: ${{ secrets.SLACK_WEBHOOK_URL }}
+
+      - name: "integration(wfb): confirm a payload was sent"
+        run: test -n "${{ steps.slackWorkflow.outputs.time }}"
+
+      - name: "integration(incoming): post a message via incoming webhook"
```

Comment:
The current method requires SLACK_WEBHOOK_TYPE: INCOMING_WEBHOOK to use incoming webhooks. This is verbose, but it's clearer to check if this action step is either a workflow webhook trigger or incoming webhooks for messaging. Can we consider something similar (like webhook_type: workflow_trigger|incoming_webhooks?) to make it more descriptive?

Reply:
Thanks for calling this out! I agree that it's super helpful when making edits to an action 📝

Some wording was changed to make searching a bit easier for these specific terms, but here's the current change:

```yaml
- name: Start a Slack workflow using a webhook URL
  uses: slackapi/slack-github-action
  with:
    webhook: ${{ secrets.SLACK_WEBHOOK_URL }}
    webhook-type: incoming-webhook  # https://api.slack.com/messaging/webhooks
    webhook-type: webhook-trigger   # https://api.slack.com/automation/triggers/webhook
```

With either one of those last options being required with `webhook` 🔗